datreant/backends/core.py

```diff
@@ -316,3 +313,115 @@ def _close(self):
         self._unlock(self.fd)
         self.fdlock = None
         self._close_fd()
+
+
+class FileSerial(File):
+    """File object base class for serialization formats, such as JSON.
+
+    """
+    def _open_file_r(self):
+        return open(self.filename, 'r')
+
+    def _open_file_w(self):
+        return open(self.filename, 'w')
+
+    @staticmethod
+    def _read(func):
+        """Decorator for opening state file for reading and applying shared
+        lock.
+
+        Applying this decorator to a method will ensure that the file is opened
+        for reading and that a shared lock is obtained before that method is
+        executed. It also ensures that the lock is removed and the file closed
+        after the method returns.
+
+        """
+        @wraps(func)
+        def inner(self, *args, **kwargs):
+            if self.fdlock:
```

Comment:
Done. Since these decorators are always used in a particular way, there was no use having four of them. There are now only `_read` and `_write`.